docs/pages/guides/config-plugins.md

```diff
@@ -610,7 +686,86 @@ Notice that `expo-location` uses `version: '11.0.0'`, and `react-native-maps` us
-For instance, say you have an `UNVERSIONED` Facebook plugin in your project, if the Facebook native code or plugin has a breaking change, that will break the way your project ejects and cause it to error on build.
+For instance, say you have an `UNVERSIONED` Facebook plugin in your project, if the Facebook native code or plugin has a breaking change, that will break the way your project prebuilds and cause it to error on build.
+
+## Static Modification
+
+Plugins can technically be used to regex file changes into application code, but these modifications are dangerous, if the template changes over time then the regex becomes hard to predict (similarly, if the user modifies a file manually or uses a custom template). Here are some examples of files you shouldn't modify manually, and alternatives.
+
+### Android Gradle Files
+
+Gradle files are written in either groovy or kotlin, they are used to manage dependencies, versioning, and other settings in the Android app. Instead of modifying them directly with `withProjectBuildGradle`, `withAppBuildGradle`, or `withSettingsGradle`, utilize the static `gradle.properties` file.
+
+The `gradle.properties` is a static key/value pair that groovy files can read from. For example, say you wanted to control some toggle in groovy:
+
+`gradle.properties`
+
+```
+/* @info Safely modified using the <code>withGradleProperties()</code> mod. */
+expo.react.jsEngine=hermes
+/* @end */
+```
+
+Then later in a gradle file:
+
+`app/build.gradle`
+
+```groovy
+project.ext.react = [
+  /* @info This code would be added to the template ahead of time, but it could be regexed in using <code>withAppBuildGradle()</code> */
+  enableHermes: findProperty('expo.react.jsEngine') ?: 'jsc'
+  /* @end */
+]
+```
+
+- For keys in the `gradle.properties`, we use camel case separated by `.`s, and usually starting with the `expo` prefix to denote that the property is managed by prebuild.
+- To access the property, use one of two global methods:
+  - `property`: Get a property, throw an error if the property is not defined.
+  - `findProperty`: Get a property without throwing an error if the property is missing. This can often be used with the `?:` operator to provide a default value.
+
+Generally, you should only interact with the Gradle file via Expo [Autolinking][autolinking], this provides a programmatic interface with the project files.
+
+### iOS App Delegate
+
+Some modules may need to add delegate methods to the project AppDelegate, this can done dangerously via the `withAppDelegate` mod, or it can be done safely by adding support for unimodules AppDelegate proxy to the native module. The unimodules AppDelegate proxy can swizzle function calls to native modules in a safe and reliable way. If the language of the project AppDelegate changes from Objective-c to Swift, the swizzler will continue to work, whereas a regex would possibly fail.
```

Comment:
```suggestion
Some modules may need to add delegate methods to the project AppDelegate, this can done dangerously via the `withAppDelegate` mod, or it can be done safely by adding support for unimodules AppDelegate proxy to the native module. The unimodules AppDelegate proxy can swizzle function calls to native modules in a safe and reliable way. If the language of the project AppDelegate changes from Objective-C to Swift, the swizzler will continue to work, whereas a regex would possibly fail.
```